### PR TITLE
Add a dart:html client

### PR DESCRIPTION
**Pull request by donny-dont**
Adding the browser_client.dart implementation

### CODE REVIEW

File: lib/browser_client.dart [hunk 1]
```diff
@@ -41,16 +41,17 @@ class BrowserClient extends BaseClient {
   bool withCredentials = false;
 
   /// Sends an HTTP request and asynchronously returns the response.
-  Future<StreamedResponse> send(BaseRequest request) async {
-    var bytes = await request.finalize().toBytes();
+  Future<Response> send(Request request) async {
+    var stream = new ByteStream(request.read());
```
Comment (nex3) [inline lib/browser_client.dart]: I really want to kill `ByteStream`, so we shouldn't use it here. You can pull this functionality into a function in `utils/`, although really it should live in a package somewhere.

**Comment by donny-dont** [comment]
The `responseUrl` does not appear to need any modification. Its the absolute path from what I see in testing.

File: lib/browser_client.dart [hunk 2]
```diff
@@ -59,13 +60,12 @@ class BrowserClient extends BaseClient {
...
-            new ByteStream.fromBytes(body),
+        completer.complete(new Response(
+            xhr.responseUrl,
             xhr.status,
-            contentLength: body.length,
-            request: request,
-            headers: xhr.responseHeaders,
-            reasonPhrase: xhr.statusText));
+            reasonPhrase: xhr.statusText,
+            body: new ByteStream.fromBytes(body),
```
Comment (nex3) [inline lib/browser_client.dart]: This also shouldn't be a `ByteStream`. In this case, just `new Stream.fromIterable([body])` should work.

File: lib/browser_client.dart (hunk 1 above)
Comment (donny-dont) [inline lib/browser_client.dart]: It appears it IS in a package 👍

File: lib/browser_client.dart (hunk 2 above)
Comment (donny-dont) [inline lib/browser_client.dart]: Fixed

**Comment by donny-dont** [comment]
@nex3 :shipit: ?

**Comment by donny-dont** [comment]
Nice! Another one down 😄

**Pull request merged**